prog/vnet/load_balancer_nexus.rb

```diff
@@ -104,6 +104,10 @@ def need_to_rewrite_dns_records?
     reap
     if strand.children.select { _1.prog == "Vnet::CertServer" }.all? { _1.exitval == "certificate is reshared" } || strand.children.empty?
       decr_refresh_cert
+      load_balancer.certs_dataset.exclude(id: load_balancer.active_cert.id).all do |cert|
+        cert.incr_destroy
+        CertsLoadBalancers[cert_id: cert.id].destroy
```

Comment:
`CertsLoadBalancers#destroy` already calls `cert.incr_destroy`.  This results in `cert.incr_destroy` being called twice.  Is that what we want?

Additionally, `CertsLoadBalancers[cert_id: cert.id]` could return an entry for a different load balancer, if the certificate is associated to multiple load balancers.  This should probably be `CertsLoadBalancers[cert_id: cert.id, load_balancer_id: load_balancer.id]` instead.

Reply:
IIRC there is no such thing as a shared certificate at present since they're keyed to the load balancer's DNS name, which is essential to the whole thing. But it may not hurt to program defensively.

Reply:
Interesting.  Looks like I misunderstood the purpose of `CertsLoadBalancers`?  It looks like a join table between certs and load balancers.  However, I see now that we have `Cert.one_through_one :load_balancer` and not many_to_many .

I guess its done this way because you can have multiple certificates to a load balancer, but not the other way around, and you don't want to have the load_balancer_id in the certificates table because not all certs are tied to a load balancer?

Is the double `cert.incr_destroy` still an issue?

Reply:
It may be wasteful and unnecessary (depending if the other site for incr_destroy has other entrants), but it's not harmful. But if it's possible to eliminate one and be less confusing, that would be better.

What is harmful is an incr in a loop, because it generates one tuple per iteration

Reply:
You can see the older system (now much older, I suppose) here: https://www.citusdata.com/blog/2016/08/12/state-machines-to-run-databases/

The current record-per-incr is effectively a unary pivot of what used to be an integer, to better be able to attach data to each increment, and to have finer grained locking data. But, it blows up the amount of state per increment by billions of times (not an exaggeration, considering previously, tweaking a bit), so it's definitely pushing an envelope of some kind.

Reply:
The main reason CertsLoadBalancers table exist is that we would like to use Certs in future for other products as well, such as Postgres. In such a case, we would need to add a new columnt to the Cert table for every new service. So, I chose to implement a join table

Reply:
Even though it's not harmful, it seems best to remove the `cert.incr_destroy` in this code, since it's in `CertsLoadBalancers#destroy`.  However, as @fdr says it isn't harmful, it doesn't block approval.

Reply:
Oh, I have removed it already